tests/integration_tests/test_sklearn.py

```diff
@@ -29,7 +32,12 @@ def test_optuna_search(enable_pruning):
     with pytest.raises(NotFittedError):
         optuna_search._check_is_fitted()
 
-    optuna_search.fit(X, y)
+    if fit_params == 'sample_weight':
+        sample_weight = np.ones_like(y)
+        optuna_search.fit(X, y, sample_weight=sample_weight)
```

Comment:
`sample_weight` won't work (shape mismatch error, observed in CI) as we're doing a CV and the number of samples will vary. How about testing `fit_params` only when pruning is disabled as follows (when pruning is enabled `fit_params` will be passed to both `fit` and `partial_fit` it seems).

```python
@pytest.mark.parametrize('enable_pruning', [True, False])
@pytest.mark.parametrize('fit_params', ['', 'sample_weight'])
@pytest.mark.filterwarnings('ignore::UserWarning')
def test_optuna_search(enable_pruning, fit_params):
    # type: (bool, str) -> None

    X, y = make_blobs(n_samples=10, centers=3)
    est = SGDClassifier(max_iter=5, tol=1e-03, warm_start=True)
    ...
    optuna_search = integration.OptunaSearchCV(
        est,
        param_dist,
        cv=3,
        enable_pruning=enable_pruning,
        error_score='raise',
        max_iter=5,
        random_state=0,
        return_train_score=True
    )
    ....
    if fit_params == 'sample_weight' and not enable_pruning:
        # `SGDClassifier.fit`, not `SGDClassifier.partial_fit`.
        optuna_search.fit(X, y, coef_init=np.ones((3, 2), dtype=np.float64))
    else:
        optuna_search.fit(X, y)
    ...
```

Reply:
It makes sense, thanks.
I revised my PR to use `coef_init` in 282f883.